clients/hadoopfs/src/main/java/io/lakefs/committer/DummyOutputCommitter.java

```diff
@@ -0,0 +1,111 @@
+package io.lakefs.committer;
+
+import io.lakefs.LakeFSFileSystem;
+import io.lakefs.utils.ObjectLocation;
+
+import org.apache.commons.lang.exception.ExceptionUtils; // (for debug prints ONLY)
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.JobStatus;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.TaskAttemptID;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputCommitter;
+
+import com.google.common.base.Preconditions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+// TODO(ariels): For Hadoop 3, it is enough (and better!) to extend PathOutputCommitter.
+public class DummyOutputCommitter extends FileOutputCommitter {
+    private static final Logger LOG = LoggerFactory.getLogger(DummyOutputCommitter.class);
+
+    protected String branch = null;
+    protected String outputBranch = null;
+    protected Path outputPath = null;
+    protected String workBranch;
+    protected Path workPath;
+
+    public DummyOutputCommitter(Path outputPath, JobContext context) throws IOException {
+        super(outputPath, context);
+        // TODO(lynn): Create lakeFS API client.
+
+        if (outputPath != null) {
+            FileSystem fs = outputPath.getFileSystem(context.getConfiguration());
+            Preconditions.checkArgument(fs instanceof LakeFSFileSystem,
+                                        "%s not on a LakeFSFileSystem", outputPath);
+            ObjectLocation outputLocation = ObjectLocation.pathToObjectLocation(null, outputPath);
+            this.outputBranch = outputLocation.getRef();
+            this.outputPath = fs.makeQualified(outputPath);
+        }
+    }
+
+    public DummyOutputCommitter(Path outputPath, TaskAttemptContext context) throws IOException {
+        this(outputPath, (JobContext)context);
+
+        TaskAttemptID id = context.getTaskAttemptID();
+        // TODO(ariels): s/[^-\w]//g on the branch ID, ensuring all chars
+        // are allowed.  (Some) users might manage to create a bad task
+        // name.
+        this.branch = id.toString();
+        Preconditions.checkNotNull(outputPath, "missing outputPath");
```

Comment:
outputPath [can be null](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/mapreduce/lib/output/FileOutputCommitter.html#FileOutputCommitter-org.apache.hadoop.fs.Path-org.apache.hadoop.mapreduce.TaskAttemptContext-:~:text=Constructor%20Detail-,FileOutputCommitter,-public%C2%A0FileOutputCommitter())